cmd/syncer/main.go

```diff
@@ -61,9 +60,9 @@ func main() {
 	ctx, log := logger.GetNewContextWithLogger()
 	log.Infof("Version : %s", syncer.Version)
 
-	clusterFlavor := cnstypes.CnsClusterFlavor(os.Getenv(csitypes.EnvClusterFlavor))
-	if clusterFlavor == "" {
-		clusterFlavor = cnstypes.CnsClusterFlavorVanilla
+	clusterFlavor, err := config.GetClusterFlavor(ctx)
+	if err != nil {
+		log.Errorf("Failed retrieving cluster flavor. Error: %v", err)
```

Comment:
Dont you want to return/exit if cluster flavour is empty?

Reply:
`clusterFlavor` is only empty when `err != nil`. As this is the main function, I noticed that we do not return error as it might crash the container, so I have not returned the error, just logged it.

Reply:
Yes, it will be empty. Do we want to default to `Vanilla`?